src/dragonball/src/dbs_virtio_devices/src/vhost/vhost_user/block.rs

```diff
@@ -0,0 +1,784 @@
+        } else {
+            for idx in 0..device.queue_sizes.len() {
+                let event = Events::with_data(
+                    &device.intr_evts[idx],
+                    QUEUE_AVAIL_SLOT + idx as u32,
+                    EventSet::IN,
+                );
+                if let Err(err) = ops.add(event) {
+                    error!(
+                        "{}: failed to register epoll event for queue {}, {:?}",
+                        self.id, idx, err
+                    );
+                }
+            }
+        }
+    }
+}
+
+struct VhostUserBlockDevice {
+    vhost_socket: String,
+    queue_sizes: Arc<Vec<u16>>,
+    device_info: VirtioDeviceInfo,
+    endpoint: Endpoint,
+    intr_evts: Arc<Vec<EventFd>>,
+    timer_fd: Option<TimerFd>,
+    curr_queues: u32,
+    id: String,
+}
+
+impl VhostUserBlockDevice {
+    pub fn new(
+        config_path: String,
+        queue_sizes: Arc<Vec<u16>>,
+        event_mgr: EpollManager,
+    ) -> VirtIoResult<Self> {
+        // config_path = "spdk://xxxxxxx.sock", remove the prefix "spdk://"
+        let vhost_socket = String::from(&config_path[7..]);
+
+        let init_queues = queue_sizes.len() as u32;
+
+        info!("vhost-user-blk: try to connect to {:?}", vhost_socket);
+        // Connect to the vhost-user socket.
+        let mut master = Master::connect(&vhost_socket, 1).map_err(VirtIoError::VhostError)?;
+
+        info!("vhost-user-blk: get features");
+        let avail_features = master.get_features().map_err(VirtIoError::VhostError)?;
+        info!(
+            "vhost-user-blk: get features done, ret:{:?}, queue_size: {:?}",
+            avail_features,
+            queue_sizes.len()
+        );
+
+        // for the standard vhost_user_blk device, get the device config from slave.
+        let config_space = {
+            master.set_features(avail_features)?;
+            let protocol_featuers = master.get_protocol_features()?;
+            // set the config features to get the device's config from slave.
+            master.set_protocol_features(protocol_featuers)?;
+
+            let config_len = mem::size_of::<VirtioBlockConfig>();
+            let config_space: Vec<u8> = vec![0u8; config_len as usize];
+
+            let (_, mut config_space) = master
+                .get_config(
+                    VHOST_USER_CONFIG_OFFSET,
+                    config_len as u32,
+                    VhostUserConfigFlags::WRITABLE,
+                    config_space.as_slice(),
+                )
+                .map_err(VirtIoError::VhostError)?;
+
+            // set the num queues
+            config_space[CONFIG_SPACE_SIZE - 2] = init_queues as u8;
+            config_space[CONFIG_SPACE_SIZE - 1] = (init_queues >> 8) as u8;
+            config_space
+        };
+
+        let mut intr_evts = Vec::with_capacity(init_queues as usize);
+        for _ in 0..init_queues {
+            intr_evts.push(EventFd::new(0).map_err(VirtIoError::IOError)?);
+        }
```

Comment:
Does the following code work?

```suggestion
        let mut intr_evts = vec![EventFd::new(0).map_err(VirtIoError::IOError)?; init_queues as usize];
```

Reply:
This works:
let intr_evts: Vec<EventFd> = (0..init_queues).map(|_| EventFd::new(0).unwrap()).collect();